packages/core/src/dialects/abstract/query-generator.js

```diff
@@ -1726,7 +1726,8 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
   }
 
   _getAliasForField(tableName, field, options) {
-    if (this.options.minifyAliases && options.aliasesByTable[`${tableName}${field}`]) {
+    const shouldMinifyAlias = options?.minifyAliases ?? this.options.minifyAliases;
+    if (shouldMinifyAlias && options.aliasesByTable[`${tableName}${field}`]) {
```

Comment:
```suggestion
    if (options.minifyAliases && options.aliasesByTable[`${tableName}${field}`]) {
```
Not very sure why but changing this fixes the failing integrations tests in at least the following test suite; https://github.com/sequelize/sequelize/blob/main/packages/core/test/integration/include/limit.test.js

Haven't tested it on the entire suite yet

Reply:
I think it's because of that null coalesce, that `??` operator, checks for `null` which I think from somewhere I am passing as `false` or probably is being coerced into `false` therefore `this.options.minifyAliases` never kicks in even if `this.options.minifyAliases` is true! Thanks a lot for this insight @WikiRik